snooty/postprocess.py

```diff
@@ -413,6 +416,94 @@ def get_include_argument(node: n.Directive) -> str:
             assert isinstance(ast, n.Parent)
             node.children = ast.children
 
+        elif node.name == "literalinclude":
+            argument = get_include_argument(node)
+            include_slug = clean_slug(argument)
+            include_filepath = (
+                self.project_config.root / self.project_config.source / include_slug
+            )
+
+            # Attempt to read the literally included file
+            try:
+                with open(include_filepath) as file:
+                    text = file.read()
+
+            except OSError as err:
+                self.diagnostics[filename].append(
+                    CannotOpenFile(include_filepath, err.strerror, node.span + (-1,))
+                )
+                return
```

Comment:
The old `PendingLiteralInclude` implementation uses this [`add_static_asset()`](https://github.com/mongodb/snooty-parser/blob/master/snooty/parser.py#L665) function, which takes care of generating the correct path and loading the file in a safe manner. Perhaps you can try using `reroot_path()` and `StaticAsset.load()` here to leverage some of our existing code!